### PR TITLE
[#68] detect rectified cases

### PR DESCRIPTION
**Pull request by TaimoorRana**
**Description**
As a user, I want to see which cases were rectified, appealed, etc.

**Task completed:** 

1.  Process text files in a given directory
2.  Detects whether a case had multiple hearings
3.  Differentiates between the original case and a rectified case with their different "No dossier"
4.  Parse a user-defined amount of randomly selected cases
5.  Save extracted data in JSON format

**Sample JSON output:**
`{"noDossier": ["123456 12 12345678 D"], "noDemande": ["1234567"], "isRectified": true, "totalHearings": 2}`

### CODE REVIEW

File: src/ml_service/feature_extraction/dummy_cases/sample_case1.txt [hunk 1]
```diff
@@ -0,0 +1 @@
...





No dossier:
"NODOSSIER"
No demande:
"NODEMANDE"
Page 2
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/dummy_cases/sample_case1.txt]: This is fake data. All names, dates, and amounts are fake with most text replaced with lorem lipsum.

**Review by arekmano — APPROVED** [review]
LGTM

File: src/ml_service/feature_extraction/case.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import re
+import codecs
+import json
+import os
+
+
+class Case:
+    def __init__(self, file_path):
+        self.data = {"noDossier": 0, "noDemande": 0, "isRectified": False, "totalHearings": 0}
```
Comment (Vynny) [inline src/ml_service/feature_extraction/case.py]: Do the keys have to be French? Why not recordNumber, caseNumber..etc?

**Review by Vynny — APPROVED** [review]
Looks good to me, was only confused with the French keys names

File: src/ml_service/feature_extraction/case.py (hunk 2 above)
Comment (TaimoorRana) [reply src/ml_service/feature_extraction/case.py]: They can be in English for sure, I have no preference. I’ll change it.

Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/case.py]: yes english plz

**Pull request merged**